src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -269,7 +274,7 @@ - (void)sendHeaders:(NSDictionary *)headers {
   // TODO(jcanizales): Add error handlers for async failures
   GRPCOpSendMetadata *op = [[GRPCOpSendMetadata alloc] initWithMetadata:headers
                                                                   flags:[GRPCCall callFlagsForHost:_host path:_path]
-                                                                handler:nil];
+                                                                handler:nil];  // No clean-up needed after SEND_INITIAL_METADATA
```

Comment:
Let's just remove the parameter (and if you want, add the comment in `GRPCOpSendMetadata` explaining why there's no handler).